crates/strategy/zksync/src/backend/runner.rs

```diff
@@ -125,6 +129,64 @@ impl BackendStrategyRunner for ZksyncBackendStrategyRunner {
             &ZksyncMergeState { persistent_immutable_keys: &ctx.persistent_immutable_keys };
         ZksyncBackendMerge::merge_zk_account_data(addr, active, fork_db, zk_state);
     }
+
+    // Note(zk): This function in upstream code is not implemented as part of the strategy pattern,
+    // but is instead a standalone function. We have moved it here to make it part of the
+    // strategy, as there is some abstraction in the middle since the envelopes, and types are
+    // different. See new transact_from_tx evm implementation in:
+    // crates/evm/core/src/backend/strategy.rs See the main function in entrypoint:
+    // crates/evm/core/src/backend/mod.rs
+    fn transact_from_tx(
+        &self,
+        backend: &mut Backend,
+        data: &Bytes,
+        mut env: Env,
+        journaled_state: &mut JournaledState,
+        inspector: &mut dyn InspectorExt,
+    ) -> eyre::Result<TransactionMaybeSigned> {
+        let envelope: ZkTxEnvelope =
+            ZkTxEnvelope::decode_2718(&mut data.as_ref()).wrap_err("Failed to decode tx")?;
+
+        let tx_712 = envelope.as_eip712();
+        let parts = tx_712.unwrap().clone().into_parts().0;
+
+        let tx: TransactionRequest = TransactionRequest {
+            from: Some(parts.from),
+            max_fee_per_gas: Some(parts.max_fee_per_gas),
+            max_priority_fee_per_gas: Some(parts.max_priority_fee_per_gas),
+            max_fee_per_blob_gas: Default::default(),
+            gas: Some(parts.gas),
+            input: Some(parts.input).into(),
+            chain_id: Some(parts.chain_id),
+            access_list: Default::default(),
+            transaction_type: Default::default(),
+            blob_versioned_hashes: Default::default(),
+            sidecar: Default::default(),
+            authorization_list: Default::default(),
+            to: Some(alloy_primitives::TxKind::Call(parts.to)),
+            value: Some(parts.value),
+            nonce: Some(journaled_state.state.get(&parts.from).unwrap().info.nonce),
```

Comment:
This isn't present in the encoded tx?
Does upstream also do a similar edit or it assumes the right nonce is encoded?

Reply:
nope... see: https://github.com/matter-labs/foundry-zksync/pull/958/commits/69cd80b38c7be219da87634853fe58926dad23a7
This is what @Jrigada fixed, it comes from the journaled_state